File: android/src/main/java/com/kiki/react/amap/search/AMapSearchPackage.java

package com.rnplayground;

import com.facebook.react.ReactPackage;
import com.facebook.react.bridge.NativeModule;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.uimanager.ViewManager;
import com.rnplayground.cloud.AMapCloudSearch;
import com.rnplayground.cloud.AMapPoiSearch;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;

public class AMapSearchPackage implements ReactPackage {

    @Override
    public List<NativeModule> createNativeModules(ReactApplicationContext reactContext) {
        return Arrays.<NativeModule>asList(
                new AMapCloudSearch(reactContext),
                new AMapPoiSearch(reactContext)
        );
    }

    @Override
    public List<ViewManager> createViewManagers(ReactApplicationContext reactContext) {
        return Collections.emptyList();
    }
}

File: lib/android/src/main/java/com/kiki/react/amap/search/AMapSearchPackage.kt
package com.kiki.react.amap.search

import com.facebook.react.ReactPackage
import com.facebook.react.bridge.NativeModule
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.uimanager.ViewManager
import com.kiki.react.amap.search.cloud.CloudSearchModule
import com.kiki.react.amap.search.cloud.GeocodeModule
import com.kiki.react.amap.search.cloud.PoiSearchModule

class AMapSearchPackage : ReactPackage {
    override fun createNativeModules(reactContext: ReactApplicationContext): List<NativeModule> {
        return listOf(
                CloudSearchModule(reactContext),
                PoiSearchModule(reactContext),
                GeocodeModule(reactContext)
        )
    }

    override fun createViewManagers(reactContext: ReactApplicationContext): List<ViewManager<*, *>> {
        return emptyList()
    }

}
